tests/aws/services/s3/test_s3_list_operations.py

```diff
@@ -46,6 +48,87 @@ def assert_timestamp_is_iso8061_s3_format(timestamp: str):
     assert parsed_ts.microsecond == 0
 
 
+class TestS3ListBuckets:
+    @markers.aws.validated
+    def test_list_buckets_by_prefix_with_case_sensitivity(
+        self, s3_create_bucket, aws_client, snapshot
+    ):
+        snapshot.add_transformer(snapshot.transform.s3_api())
+
+        bucket_name = f"test.bucket.{short_uid()}"
+        s3_create_bucket(Bucket=bucket_name)
+        s3_create_bucket(Bucket=f"ignored.bucket.{short_uid()}")
+
+        response = aws_client.s3.list_buckets(Prefix=bucket_name.upper())
+        assert len(response["Buckets"]) == 0
+
+        response = aws_client.s3.list_buckets(Prefix=bucket_name)
+        assert len(response["Buckets"]) == 1
+
+        returned_bucket = response["Buckets"][0]
+        assert returned_bucket["Name"] == bucket_name
+
+        snapshot.match("list-objects-by-prefix", response)
+
+    @markers.aws.validated
+    @markers.snapshot.skip_snapshot_verify(paths=["$..ContinuationToken"])
```

Comment:
for values that will change every time you run the test against AWS, you will need to transform the value instead of ignoring it. The best practice when writing and running tests against AWS is to be able to re-run the test against AWS and for it to pass without updating the snapshot. 

Here, you would need to add this line, for example: `snapshot.add_transformer(snapshot.transform.key_value("ContinuationToken"))`